### PR TITLE
Portability fixes

### PR DESCRIPTION
**Pull request by 0-wiz-0**

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

**Review by MrAnno — DISMISSED** [review]
Hi,

Thank you for opening this PR.
AFAIK these are the remaining patches from the NetBSD syslog-ng packaging.

The `sigaction` dlsym trickery that we use in `signal-handler.c` has additional issues due to its unportable nature, for example, it does not work with shared object load order randomization (#3445). We are planning to reimplement this part in a more portable manner. Until then, I think this `__NetBSD__` ifdef would not hurt :)

**Comment by 0-wiz-0** [comment]
@MrAnno Yes, I took those from the pkgsrc package.
There is one additional patch but I don't know how much of that you'd want to take -- please check it out:

https://github.com/NetBSD/pkgsrc/blob/trunk/sysutils/syslog-ng/patches/patch-ac

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
Oh, I did not know about this script (it seems to be a nice syslogd to syslog-ng config converter).
The `contrib` folder is not actively maintained, but I think we could add NetBSD support to this script as well, if you use that. 
However, we need to check the `@UNAME@` part.

@alltilla What do you think?

File: lib/signal-handler.c [hunk 1]
```diff
@@ -84,12 +84,16 @@ signal_handler_exec_external_handler(gint signum)
 static int
 _original_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact)
 {
+#ifdef __NetBSD__
+  __libc_sigaction14(signum, act, oldact);
```
Comment (MrAnno) [inline lib/signal-handler.c:88]: The original sigaction result should be returned.

```suggestion
  return __libc_sigaction14(signum, act, oldact);
```

**Comment by 0-wiz-0** [comment]
On NetBSD, uname is in `/usr/bin/uname`. Could we remove the use of the path component in the script, since both /bin and /usr/bin are usually in the user's paths?
I've left out this part for now but added the uncontroversial (I hope) parts.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@0-wiz-0 Thank you for addressing everything.

I'm not an `awk` wizard, but calling `uname` without specifying the full path worked for me as well. I've seen `awk` scripts doing that.

One last thing: unfortunately, there is a failing commit check in our CI.
- We prefer signing all of our commits using git's sign-off feature. You can do that using the `--signoff` flag of the `git commit` command, or if you want to add your sign-off line to an existing commit, `git rebase` has the same flag.
- Commit `Avoid unportable test(1) operator. ` does not specify the "unit"/"module" where the change was made. For example, the following would be accepted by the commit-check job:
  - `build: avoid unportable test(1) operator` or
  - `autotools: avoid unportable test(1) operator`, etc.

**Comment by 0-wiz-0** [comment]
Thanks for the explanations, @MrAnno ! I think I've addressed all of these in the last push.

**Review by MrAnno — APPROVED** [review]
👍🏻

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by Kokan — APPROVED** [review]

**Pull request merged**